gitops/gitops.go

```diff
@@ -57,13 +59,7 @@ func GenerateAndPush(outputPath string, remote string, component appstudiov1alph
 		}
 	}
 
-	if out, err := e.Execute(repoPath, "rm", "-rf", filepath.Join("components", componentName)); err != nil {
```

Comment:
If resources change between generations, and the subsequent gitops generation don't include a resource that was present in the previous generation, then the old resources would be left behind and not removed (see below for an example)

E.g., right now, when we generate gitops resources for a given component, if a component specifies a targetPort, we'll include a route in the resources that we generate. But if a user were to remove that targetPort, the next generation wouldn't include the route resource, but since we're no blowing away the base folder.

We could either blow away the base folder on each generation (but this would cause issues with my other suggestion) or we could:
1) Read the old kustomize.yaml
2) Then parse the entries under `base` and delete each file before generation 

It's a bit of an edge case, but I think it's something we need to handle, especially once we generate gitops resources from the Kubernetes resources in the devfile.

Reply:
We are doing a git diff right? So I think it should catch the subsequent removal of targetPort, since we will not generate the route and there would be a git diff 🤔

Reply:
Hmm? But in this case, the route file won't be touched, since it's not one of the files we're generating, so it won't be included in the git diff

Reply:
git diff might only catch what has been editted. so John was right, the route.yaml wouldn't be removed in this case. 
I'm wondering if that's something affecting the behavior? removing the targetPort will remove the `route.yaml` from the kustomization file, so should not affect anything right? just a dangling `route.yaml` left in the gitops folder.

Reply:
so would it be an issue if there is a route.yaml left behind if its not mentioned in the kustomize.yaml 🤔 

the other alternative is to read the kustomize file and cross check with the files in the dir and remove any unnecessary ones but if my earlier comment makes sense ie kustomize.yaml doesnt have an entry then we can skip this..

Reply:
emm. I think it also depends this question John raised https://github.com/redhat-appstudio/application-service/pull/151#discussion_r924847171
if we allow custom patches under base folder, then we should not blow the folder or delete any files. (even it's not included in the `kustomization.yaml` file, might be a custom file temporarily removed by user, should not delete the source file). If we do not allow custom patches under base folder, then I think we can assume the base folder only contains generated files, so base folder can be safely deleted.

Reply:
Let's just blow away the base dir for now? And if we decide to allow custom patches under the base folder late, we'll handle deletion like I suggested in https://github.com/redhat-appstudio/application-service/pull/151#discussion_r924835458